### PR TITLE
Implement FE_Poly::memory_consumption 

### LINKED ISSUE #9102: FE_DGQ::memory_consumption() is not implmented
The method `FE_DGQ::memory_consumption()` is not implemented:

https://github.com/dealii/dealii/blob/fe4975c998b1fdc124aef18238e3eb06d7ab88ed/source/fe/fe_dgq.cc#L822-L828

so that the following program fails:

```cpp
#include <deal.II/dofs/dof_handler.h>

#include <deal.II/grid/grid_generator.h>
#include <deal.II/grid/tria.h>

#include <deal.II/fe/fe_dgq.h>

using namespace dealii;

const int dim    = 2;
const int degree = 2;

int main()
{
    
    if(false)
    {
      Triangulation<dim> tria;
      GridGenerator::hyper_cube(tria);
    
      FE_DGQ<dim> fe(degree);
      DoFHandler<dim> dof_handler(tria);
      dof_handler.distribute_dofs (fe);
    
      // failing
      std::cout << dof_handler.memory_consumption () << std::endl;
    }
    
    if(true)
    {
      FE_DGQ<dim> fe(degree);
      
      // failing
      std::cout << fe.memory_consumption () << std::endl;
    }
    
}
```

Is there a particular reason we cannot use the implementation from `FiniteElement`?

### PR DESCRIPTION
**Pull request by peterrum**
...so that memory_consumption works for `FE_Q` and `FE_DGQ`.

Closes #9102.

ping @bangerth

### CODE REVIEW

File: include/deal.II/base/polynomial.h [hunk 1]
```diff
@@ -225,6 +225,12 @@ namespace Polynomials
     void
     serialize(Archive &ar, const unsigned int version);
 
+    /**
+     * Return an estimate (in bytes) or the memory consumption of this object.
```
Comment (kronbichler) [inline include/deal.II/base/polynomial.h]:
```suggestion
     * Return an estimate (in bytes) for the memory consumption of this object.
```

File: include/deal.II/base/polynomials_piecewise.h [hunk 2]
```diff
@@ -132,6 +132,12 @@ namespace Polynomials
     void
     serialize(Archive &ar, const unsigned int version);
 
+    /**
+     * Return an estimate (in bytes) or the memory consumption of this object.
```
Comment (kronbichler) [inline include/deal.II/base/polynomials_piecewise.h]:
```suggestion
     * Return an estimate (in bytes) for the memory consumption of this object.
```

File: include/deal.II/base/scalar_polynomials_base.h [hunk 3]
```diff
@@ -138,6 +138,12 @@ class ScalarPolynomialsBase
   virtual std::string
   name() const = 0;
 
+  /**
+   * Return an estimate (in bytes) or the memory consumption of this object.
```
Comment (kronbichler) [inline include/deal.II/base/scalar_polynomials_base.h]:
```suggestion
   * Return an estimate (in bytes) for the memory consumption of this object.
```

File: include/deal.II/base/tensor_product_polynomials.h [hunk 4]
```diff
@@ -209,6 +209,12 @@ class TensorProductPolynomials : public ScalarPolynomialsBase<dim>
   virtual std::unique_ptr<ScalarPolynomialsBase<dim>>
   clone() const override;
 
+  /**
+   * Return an estimate (in bytes) or the memory consumption of this object.
```
Comment (kronbichler) [inline include/deal.II/base/tensor_product_polynomials.h]:
```suggestion
   * Return an estimate (in bytes) for the memory consumption of this object.
```

File: include/deal.II/fe/fe_poly.h [hunk 5]
```diff
@@ -227,6 +227,12 @@ class FE_Poly : public FiniteElement<dim, spacedim>
                                  const Point<dim> & p,
                                  const unsigned int component) const override;
 
+  /**
+   * Return an estimate (in bytes) or the memory consumption of this object.
```
Comment (kronbichler) [inline include/deal.II/fe/fe_poly.h]:
```suggestion
   * Return an estimate (in bytes) for the memory consumption of this object.
```

**Comment by masterleinad** [comment]
What about a changelog entry and a test?

File: source/base/polynomials_piecewise.cc [hunk 6]
```diff
@@ -13,9 +13,9 @@
 //
 // ---------------------------------------------------------------------
 
+#include <deal.II/base/memory_consumption.h>
 #include <deal.II/base/polynomials_piecewise.h>
 
-
```
Comment (kronbichler) [inline source/base/polynomials_piecewise.cc]: Please undo this deletion of the line.

**Comment by masterleinad** [comment]
Otherwise, this looks good to me. Thanks!

File: source/base/scalar_polynomials_base.cc [hunk 7]
```diff
@@ -32,6 +32,14 @@ ScalarPolynomialsBase<dim>::ScalarPolynomialsBase(
   // nothing to do here for now
 }
 
+template <int dim>
```
Comment (kronbichler) [inline source/base/scalar_polynomials_base.cc]:
```suggestion


template <int dim>
```

**Comment by peterrum** [comment]
@masterleinad @kronbichler I have addressed your comments! Thanks for the review. Btw. I fixed the `or`-mistake in the rest of the library.

**Pull request merged**